app/services/trns/submission_checker.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Trns
+  class SubmissionChecker
+    attr_reader :trainee, :successful
+    alias_method :successful?, :successful
+
+    VALIDATORS = [
+      { validator: PersonalDetail, progress_key: :personal_details },
+      { validator: ContactDetail, progress_key: :contact_details },
+      { validator: DegreeDetail, progress_key: :degrees },
+      { validator: Diversities::DiversityFlow, progress_key: :diversity },
+      { validator: ProgrammeDetail, progress_key: :programme_details },
+    ].freeze
+
+    class PreSubmissionError < StandardError
+    end
+
+    class << self
+      def call(trainee)
+        new(trainee).call
+      end
+    end
+
+    def initialize(trainee)
+      @trainee = trainee
+    end
+
+    def call
+      @successful = is_submission_ready?
+
+      self
+    end
+
+  private
+
+    def is_submission_ready?
+      VALIDATORS.each do |validator_hash|
+        validator = validator_hash[:validator].new(trainee)
+        progress_value = trainee.progress.public_send(validator_hash[:progress_key])
+
+        raise PreSubmissionError, validator_hash[:validator].to_s unless progress_is_completed?(validator, progress_value)
+      end
+
+      true
+    rescue PreSubmissionError
```

Comment:
hmmm what benefits do we get with raising and rescuing the exception?

Using exception as signallers tends to be bad ideas.

Reply:
discussed offline and tried a different way https://github.com/DFE-Digital/register-trainee-teachers/pull/205/commits/b6ba50b395cd99cbe50196f368de6b0325bc2e09